packages/run-protocol/src/vaultFactory/vaultKit.js

```diff
@@ -0,0 +1,73 @@
+// @ts-check
+import { makeNotifierKit } from '@agoric/notifier';
+import '@agoric/zoe/exported.js';
+import { Far } from '@endo/marshal';
+
+const { details: X } = assert;
+
+/**
+ * @typedef {{
+ * inner: InnerVault | null,
+ * }} State
+ */
+/**
+ *
+ * @param {InnerVault} innerVault
+ */
+const wrapVault = innerVault => {
+  /** @type {NotifierRecord<VaultUIState>} */
+  const { updater, notifier } = makeNotifierKit();
+
+  /** @type {State} */
+  const state = {
+    inner: innerVault,
+  };
+
+  // Throw if this wrapper no longer owns the inner vault
+  const owned = v => {
+    const { inner } = state;
+    // console.log('OUTER', v, 'INNER', inner);
+    assert(inner, X`Using ${v} after transfer`);
+    return inner;
+  };
+
+  const vault = Far('vault', {
+    getNotifier: () => notifier,
+    makeAdjustBalancesInvitation: () =>
+      owned(vault).makeAdjustBalancesInvitation(),
+    makeCloseInvitation: () => owned(vault).makeCloseInvitation(),
+    makeTransferInvitation: () => {
+      const tmpInner = owned(vault);
+      state.inner = null;
+      return tmpInner.makeTransferInvitation();
+    },
+    // for status/debugging
+    getCollateralAmount: () => owned(vault).getCollateralAmount(),
+    getCurrentDebt: () => owned(vault).getCurrentDebt(),
+    getNormalizedDebt: () => owned(vault).getNormalizedDebt(),
+  });
+  return { vault, vaultUpdater: updater };
+};
+
+/**
+ *
+ * @param {InnerVault} inner
+ * @param {Notifier<import('./vaultManager').AssetState>} assetNotifier
+ */
+export const makeVaultKit = (inner, assetNotifier) => {
+  const { vault, vaultUpdater } = wrapVault(inner);
+  const vaultKit = harden({
+    assetNotifier,
```

Comment:
The vault notifications were broken by https://github.com/Agoric/agoric-sdk/pull/4527 because they can no longer rely on one notifier. We discussed multiple designs in the RUN protocol meeting and agreed this was the best option. https://github.com/Agoric/agoric-sdk/commit/554d41ed9f9b35cd59133818e428d3055006e1ca renamed uiNotifier to make it clear to consumers that there was no longer a single notifier for UI.

I think that's the right thing for now for clarity of how the contract works. ("UI updater" is a misnomer.) Updating the UI code is queued up in https://github.com/Agoric/agoric-sdk/issues/4482 which can happen after Restival because UI is not in scope.